src/main/java/org/uikit/VideoJNI.kt

```diff
@@ -2,48 +2,38 @@ package org.uikit
 
 import android.content.Context
 import android.net.Uri
-import android.util.Log
 import android.widget.RelativeLayout
-import com.google.android.exoplayer2.*
-import com.google.android.exoplayer2.database.ExoDatabaseProvider
-import com.google.android.exoplayer2.source.ProgressiveMediaSource
-import com.google.android.exoplayer2.trackselection.DefaultTrackSelector
-import com.google.android.exoplayer2.ui.PlayerView
-import com.google.android.exoplayer2.upstream.*
-import com.google.android.exoplayer2.upstream.cache.CacheDataSink
-import com.google.android.exoplayer2.upstream.cache.CacheDataSource
-import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor
-import com.google.android.exoplayer2.upstream.cache.SimpleCache
+import android.util.Log
+import androidx.media3.common.MediaItem
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.PlaybackParameters
+import androidx.media3.common.Player
+import androidx.media3.database.ExoDatabaseProvider
+import androidx.media3.datasource.DataSource
+import androidx.media3.datasource.FileDataSource
+import androidx.media3.datasource.cache.CacheDataSink
+import androidx.media3.datasource.cache.CacheDataSource
+import androidx.media3.datasource.cache.LeastRecentlyUsedCacheEvictor
+import androidx.media3.datasource.cache.SimpleCache
+import androidx.media3.datasource.okhttp.OkHttpDataSource
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.exoplayer.SeekParameters
+import androidx.media3.exoplayer.source.ProgressiveMediaSource
+import androidx.media3.exoplayer.trackselection.DefaultTrackSelector
+import androidx.media3.exoplayer.upstream.DefaultBandwidthMeter
+import androidx.media3.ui.PlayerView
 import org.libsdl.app.SDLActivity
+import okhttp3.Cache as OkHttpCache
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
 import java.io.File
+import java.util.concurrent.TimeUnit
 import kotlin.math.absoluteValue
 
-
-@Suppress("unused")
-class AVURLAsset(parent: SDLActivity, url: String) {
-    internal val videoSource: ProgressiveMediaSource
-    private val context: Context = parent.context
-
-    init {
-        val mediaItem = MediaItem.Builder().setUri(Uri.parse(url)).build()
-
-        // ExtractorMediaSource works for regular media files such as mp4, webm, mkv
-        val cacheDataSourceFactory = CacheDataSourceFactory(
-            context,
-            512 * 1024 * 1024,
-            64 * 1024 * 1024
-        )
-
-        videoSource = ProgressiveMediaSource
-            .Factory(cacheDataSourceFactory)
-            .createMediaSource(mediaItem)
-    }
-}
-
 @Suppress("unused")
 class AVPlayer(parent: SDLActivity, asset: AVURLAsset) {
-    internal val exoPlayer: SimpleExoPlayer
-    private var listener: Player.EventListener
+    internal val exoPlayer: ExoPlayer
```

Comment:
ok I'm confused. there's media3 which contains ExoPlayer now? I thought they just renamed it 😵